contrib/babelfishpg_tsql/sql/babelfishpg_tsql.sql

```diff
@@ -3655,3 +3655,81 @@ END
+                                                                IN "@parameter_name" sys.sysname DEFAULT NULL::sys."varchar")
+AS $$
+
+BEGIN
+        SELECT 	CAST (v.column_name AS sys.sysname) AS [PARAMETER_NAME],
+				CAST (CASE
+						WHEN v.column_type = 5 THEN 4
+                        WHEN v.column_type = 3 THEN 4
+                        ELSE v.column_type END
+                     AS smallint) AS [PARAMETER_TYPE],
+        		CAST (CASE   
+						WHEN v.type_name = 'int' THEN 8
+                        WHEN v.type_name = 'nchar' THEN 10
+                        WHEN v.type_name = 'char' THEN 3
+                        WHEN v.type_name = 'date' THEN 31
+                        WHEN v.type_name = 'nvarchar' THEN 12
+                        WHEN v.type_name = 'varchar' THEN 22
+                        WHEN v.type_name = 'table' THEN 23
+                        WHEN v.type_name = 'datetime' THEN 4
+                        WHEN v.type_name = 'datetime2' THEN 33
+                        WHEN v.type_name = 'datetimeoffset' THEN 34
+                        WHEN v.type_name = 'smalldatetime' THEN 15
+						WHEN v.type_name = 'time' THEN 32
+                        WHEN v.type_name = 'decimal' THEN 5
+						WHEN v.type_name = 'numeric' THEN 5
+                        WHEN v.type_name = 'float' THEN 6
+                        WHEN v.type_name = 'real' THEN 13
+                        WHEN v.type_name = 'nchar' THEN 10
+                        WHEN v.type_name = 'flag' THEN 2
+                        WHEN v.type_name = 'money' THEN 9
+                        WHEN v.type_name = 'smallmoney' THEN 17
+                        WHEN v.type_name = 'tinyint' THEN 20
+                        WHEN v.type_name = 'smallint' THEN 16
+                        WHEN v.type_name = 'bigint' THEN 0
+                        WHEN v.type_name = 'bit' THEN 2
+						WHEN v.type_name = 'text' THEN 18
+						WHEN v.type_name = 'ntext' THEN 11
+						WHEN v.type_name = 'binary' THEN 1
+						WHEN v.type_name = 'varbinary' THEN 21
+						WHEN v.type_name = 'image' THEN 7
+                        ELSE 0 END
+                 	AS smallint) AS [MANAGED_DATA_TYPE],
+        		CAST (CASE 
+						WHEN v.type_name IN (N'nchar', N'nvarchar') AND p.max_length <> -1 THEN p.max_length / 2
+						WHEN v.type_name IN (N'char', N'varchar', N'binary', N'varbinary') AND p.max_length <> -1 THEN p.max_length
+						WHEN v.type_name IN (N'nvarchar', N'varchar', N'varbinary') AND p.max_length = -1 THEN 0
+                		WHEN v.type_name IN (N'text', N'image') THEN 2147483647
+                		WHEN v.type_name = 'ntext' THEN 1073741823
+                		ELSE NULL END AS INT) AS [CHARACTER_MAXIMUM_LENGTH],
+        		CAST(CASE 
+						WHEN v.type_name IN (N'int', N'smallint', N'bigint', N'tinyint', N'float', N'real', N'decimal', N'numeric', N'money', N'smallmoney') 
```

Comment:
check if we can simplify this condition to something like:
```
WHEN v.precision = 0 THEN NULL
ELSE v.precision
END 
```

Reply:
We can not do this as v.precision is not always 0